### DISCUSSION TITLE
blogs/230405-step-by-step-dce5/

### DISCUSSION THREAD

**Discussion by giscus** [DaoCloud/DaoCloud-docs#1282] [Announcements]
# blogs/new-dce5-free/

DaoCloud Enterprise 5.0 下一代容器化平台集大成者，引领云原生浪潮，推动数字化转型。

https://docs.daocloud.io/blogs/new-dce5-free/

**Comment by Adityayxt** [comment]
安装依赖
sudo yum install -y yum-utils device-mapper-persistent-data lvm2
这一步如果使用的环境是centos 8.x，会存在报错，需要修改yum源，否则无法完成安装
报错信息如下：
[root@localhost ~]# sudo yum install -y yum-utils device-mapper-persistent-data lvm2
Failed to set locale, defaulting to C.UTF-8
CentOS Linux 8 - AppStream                                                                                                   88  B/s |  38  B     00:00
Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

**Reply by samzong** [reply]
遇到此类问题可以采用如下解决方式，替换为国内阿里云的 YUM 源.

```bash
sudo cd /etc/yum.repos.d/
sudo mkdir bak
sudo mv CentOS-*.repo ./bak
sudo curl -o CentOS-base.repo http://mirrors.aliyun.com/repo/Centos-8.repo
sudo yum clean all
```

**Reply by Adityayxt** [reply]
contos8都会遇到该问题，已经解决，很棒

**Reply by Adityayxt** [reply]
也可以参考该方法https://www.cnblogs.com/bpzblog/p/13918199.html

**Comment by Adityayxt** [comment]
镜像拉取失败，这个问题如何解决
[root@localhost ~]# kubectl get po -n calico-system
NAME                                       READY   STATUS                  RESTARTS   AGE
calico-kube-controllers-55d6d4648b-jnjg8   0/1     Pending                 0          16h
calico-node-vj7lb                          0/1     Init:ImagePullBackOff   0          16h
calico-typha-669fbbb757-zz785              0/1     ErrImagePull            0          16h
csi-node-driver-7tqpk                      0/2     ContainerCreating       0          16h

**Reply by samzong** [reply]
部署过程中，可能会遇到镜像拉取失败的问题，错误一般为 `ImagePullBackOff`，这是因为主机所在的网络环境无法正常连接 docker.io 导致，key 采用如下办法解决：

第一步找到具体无法拉取的镜像，可直接 describe pod <pod name>，即可看到，这里以 `calico-node-vj7lb` 为例

```bash
kubectl -n calico-system describe pod calico-node-vj7lb  | grep "Failed to pull image"

Warning  Failed   7m53s (x170 over 20h)  kubelet  Failed to pull image "docker.io/calico/pod2daemon-flexvol:v3.25.1": rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/calico/pod2daemon-flexvol:v3.25.1": failed to copy: httpReadSeeker: failed open: unexpected status code https://registry-1.docker.io/v2/calico/pod2daemon-flexvol/manifests/sha256:66629150669c4ff7f70832858af28139407da59f61451a8658f15f06b9f20436: 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

```

在异常镜像不多的情况下，手工拉取即可，可以从可行的代理 docker 镜像站拉取即可，这里以 daocloud 代理镜像站为例

把 docker.io 改为 docker.m.daocloud.io ，然后使用 ctr（containerd 的管理工具）

```bash
# 拉取镜像
ctr -n k8s.io image pull docker.m.daocloud.io/calico/pod2daemon-flexvol:v3.25.1

# 重命名镜像
ctr -n k8s.io image tag docker.m.daocloud.io/calico/pod2daemon-flexvol:v3.25.1 docker.io/calico/pod2daemon-flexvol:v3.25.1
```

> 如果是使用的 docker 为容器运行时，需要使用 docker 命令

**Comment by samzong** [comment]
部署过程中，还是可能会遇到 calico 启动不了的情况，如果不想纠结，可以降级使用 flannel

## 安装 Flannel

默认安装 flannel，如果你的pod-cidr 配置是 `10.244.0.0/16`

```bash
kubectl apply -f https://github.com/flannel-io/flannel/releases/latest/download/kube-flannel.yml
```

如果 是自定义的 pod-cidr

```bash
# Needs manual creation of namespace to avoid helm error
kubectl create ns kube-flannel
kubectl label --overwrite ns kube-flannel pod-security.kubernetes.io/enforce=privileged

helm repo add flannel https://flannel-io.github.io/flannel/
helm install flannel --set podCidr="192.168.0.0/16" --namespace kube-flannel flannel/flannel
```

## 卸载 Calico

正常情况下，安装 Calico 会使用 2 个文件

> 安装过程

安装官方的 calico-operator ，然后再部署对应的模块
```bash
kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.25.1/manifests/tigera-operator.yaml

```

```bash
# 下载配置文件模板
curl -LO https://raw.githubusercontent.com/projectcalico/calico/v3.25.1/manifests/custom-resources.yaml
grep cidr custom-resources.yaml #打印出默认的CIDR

# 确保 calico 配置的 cidr 与 kubernetes 安装 pod-cidr 保持一致
vim custom-resources.yaml
kubectl apply -f custom-resources.yaml
```

> 卸载过程

通常情况下，如果只是使用 `kubectl delete` 删除原先的 2 个文件，K8s 的 pod 还是不能成功起来，错误如下：

```log
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "58b5b7e0b64bf3fa6ca289be783d1ba663b0f9873b4a3b8f00e00a7aabbe96a2": plugin type="calico" failed (add): error getting ClusterInformation: connection is unauthorized: Unauthorized
```

这是应为系统存在 cni 的配置文件残留导致 ，需要删除如下文件

```bash
sudo rm -rf /etc/cni/net.d/*calico*
sudo rm -f /opt/cni/bin/calico*
sudo rm -f /usr/local/bin/calico*
sudo rm -rf /var/lib/cni/networks/calico/ 
```

**Comment by FloatXD** [comment]
执行安装中的./dce5-installer install-app -z -k  应改为./dce5-installer install-app -z
